mev-build-rs/src/reth_builder/reth_compat.rs

```diff
@@ -58,7 +113,54 @@ pub(crate) fn to_execution_payload(value: &SealedBlock) -> ExecutionPayload {
         base_fee_per_gas: ssz_rs::U256::from(header.base_fee_per_gas.unwrap_or_default()),
         block_hash: to_bytes32(hash),
         transactions: TryFrom::try_from(transactions).unwrap(),
-        withdrawals: TryFrom::try_from(withdrawals).unwrap(),
+        withdrawals: TryFrom::try_from(withdrawals.clone()).unwrap(),
+        blob_gas_used: header.blob_gas_used.unwrap(),
+        excess_blob_gas: header.excess_blob_gas.unwrap(),
     };
-    ExecutionPayload::Capella(payload)
+
+    let blob_tx_hashes = value.blob_transactions().iter().map(|t| t.hash()).collect::<Vec<_>>();
+    let blobs_bundle = get_blob_bundles(pool, blob_tx_hashes);
+    let deneb_payload_bundles =
+        deneb::ExecutionPayloadAndBlobsBundle { execution_payload: deneb_payload, blobs_bundle };
+
+    ExecutionPayload::Deneb(deneb_payload_bundles)
+}
+
+fn get_blob_bundles<Pool>(pool: &Pool, tx_hashes: Vec<H256>) -> BlobsBundle
```

Comment:
this logic would be encapsulated in the `PayloadBuilder` abstraction so we wouldn't need to pass this `Pool` generic around through all of this code